hyperspy/drawing/spectrum.py

```diff
@@ -339,6 +341,8 @@ def close(self):
         if self.text and self.text in self.ax.texts:
             self.ax.texts.remove(self.text)
         self.axes_manager.disconnect(self.update)
+        if self.spectrum_figure:
+            self.spectrum_figure.ax_lines.remove(self)
```

Comment:
Thank you for reporting the issue. I have just committed a fix.

I know that the fix is ugly. Hopefully we'll refactor the drawing code once #423 is merged.